src/main/java/org/elasticsearch/index/query/QueryParseContext.java

```diff
@@ -285,9 +285,8 @@ public void addInnerHits(String name, InnerHitsContext.BaseInnerHits context) {
         innerHitsContext.addInnerHitDefinition(name, context);
     }
 
-    @Nullable
-    public Query parseInnerQuery() throws IOException, QueryParsingException {
-        // move to START object
+    public QueryBuilder toQueryBuilder() throws IOException {
```

Comment:
Eventually this should become `parseInnerQuery()` once the old one can be deleted, I'm happy with any intermediate name. `parseNestedQuery()` for now? Will do javadocs. So used to there beeing none from reading rest of the code...